node-tests/helpers/generate-fake-package-manifest.js

```diff
@@ -1,13 +1,22 @@
 var fs = require('fs');
+var path = require('path');
 
 module.exports = function generateFakePackageManifest(name, version) {
-  if (!fs.existsSync('node_modules')) {
-    fs.mkdirSync('node_modules');
-  }
-  if (!fs.existsSync('node_modules/' + name)) {
-    fs.mkdirSync('node_modules/' + name);
-  }
-  fs.writeFileSync('node_modules/' + name + '/package.json', JSON.stringify({
+  let targetDir = 'node_modules/' + name;
```

Comment:
1. Do you mind adding a note to the PR explaining why this needed to change?
2. I *believe* this can just be handled by checking for the full path with `fs.existsSync` and then using `fs.mkdirSync(path, { recursive: true })` if it doesn't exist.

Everything else in this PR looks good; sorry I hadn't gotten to this sooner!

Reply:
I had to do this change to support creating scoped packages such as "@glimmer/component". This is used in tests to add glimmer component package as a dependency.

I can give it a try to refactor to use `recursive: true`.

Reply:
Pushed a commit changing to `recursive: true`. It seem to work in node v10. Let's see if tests are happy in node v8 (they should).

Reply:
First, thanks for the explanation—makes sense!

Second: alas, that seems like it's not working precisely *because* it was only added in 10.12.0—my bad. Feel free to force push back to previous, and we can merge it!

Reply:
Not a problem! Pushed, we should be good now.

Reply:
Up to the two of you whether it's worth more churn at this point, but we use `fs-extra` in a bunch of places in ember-cli-typescript already to help with this kind of thing. I think its [`ensureDirSync`](https://github.com/jprichardson/node-fs-extra/blob/HEAD/docs/ensureDir-sync.md) would fit the bill here?

Reply:
I was thinking similarly. My inclination here is: let’s land this PR, and then tighten this up a bit—using `fs-extra`, using `path` functions instead of strings which include `/`, etc.—since that’s more “oh, we notice this needs some improvement” than directly related to this task.